src/write_gcm.c

```diff
@@ -2147,6 +2176,53 @@ static monitored_resource_t *wg_monitored_resource_create(
   return NULL;
 }
 
+static monitored_resource_t *wg_monitored_resource_create_from_metadata(
+    const char *metadata_agent_url, const char *metadata_agent_port,
+    const char *project_id, const char *host) {
```

Comment:
Technically, `host` in this case is really the resource id, so how about naming it `resource_id`?